src/Microsoft.PowerShell.Commands.Utility/commands/utility/XmlCommands.cs

```diff
@@ -351,10 +427,104 @@ protected override void ProcessRecord()
+        /// </summary>
+        protected override void StopProcessing()
+        {
+            base.StopProcessing();
+            _helper.Stop();
+        }
+    }
+
+    /// <summary>
+    /// Implements ConvertFrom-Clixml command
+    /// </summary>
+    [Cmdlet(VerbsData.ConvertFrom, "Clixml", SupportsPaging = true)]
+    public sealed class ConvertFromClixmlCommand : PSCmdlet, IDisposable
+    {
+        #region Command Line Parameters
+
+        /// <summary>
+        /// gets or sets the InputString property
+        /// </summary>
+        [Parameter(Mandatory = true, Position = 0, ValueFromPipeline = true)]
+        public string InputObject { get; set; }
+
+        /// <summary>
+        /// inputObjectBuffer buffers all InputObjet contents available in the pipeline.
+        /// </summary>
+        private List<string> _inputObjectBuffer = new List<string>();
+
+        #endregion Command Line Parameters
+
+        #region IDisposable Members
+
+        private bool _disposed = false;
+
+        /// <summary>
+        /// public dispose method
+        /// </summary>
+        public void Dispose()
+        {
+            if (!_disposed)
+            {
+                if (_helper != null)
+                {
+                    _helper.Dispose();
+                    _helper = null;
+                }
+                _disposed = true;
+                GC.SuppressFinalize(this);
+            }
+        }
+
+        #endregion
+
+        private ImportXmlHelper _helper;
+
+        /// <summary>
+        /// ProcessRecord overload
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            _inputObjectBuffer.Add(InputObject);
+        }
+
+        /// <summary>
+        /// the main execution method for the convertfrom-clixml command
+        /// </summary>
+        protected override void EndProcessing()
+        {
+            // When Input is provided through pipeline, the input can be represented in the following two ways:
+            // 1. Each input in the collection is a complete clixml content. There can be multiple inputs of this format.
+            // 2. The complete input is a collection which represents a single clixml content. This is typically the majority of the case.
+            bool successfullyConverted = false;
+
+            try
+            {
+                foreach (var item in _inputObjectBuffer)
+                {
+                    _helper = new ImportXmlHelper(this, ImportXmlHelper.CreateMemoryStream(item));
                     _helper.Import();
                 }
             }
+            catch (XmlException)
+            {
+                successfullyConverted = false;
+            }
+
+            if (!successfullyConverted && _inputObjectBuffer.Count > 0)
+            {
+                _helper = new ImportXmlHelper(this, ImportXmlHelper.CreateMemoryStream(string.Join(System.Environment.NewLine, _inputObjectBuffer)));
+                _helper.Import();
```

Comment:
1. We already create `_helper` - can we reuse it?
2. Can `XmlException` be raised here? If so do we catch it and throw use freadly exception?